src/algebra/gcd_monoid.lean

```diff
@@ -340,6 +348,64 @@ begin
     exact dvd_gcd hn'k hn' }
 end
 
+theorem gcd_pow_right_dvd_pow_gcd {a b : α} {k : ℕ} : gcd a (b ^ k) ∣ (gcd a b) ^ k :=
+begin
+  by_cases hg : gcd a b = 0,
+  { rw gcd_eq_zero_iff at hg,
+    rcases hg with ⟨rfl, rfl⟩,
+    simp },
+  { induction k with k hk, simp,
+    rw [pow_succ, pow_succ],
+    transitivity gcd a b * gcd a (b ^ k),
+    apply gcd_mul_dvd_mul_gcd a b (b ^ k),
+    refine (mul_dvd_mul_iff_left hg).mpr hk }
+end
+
+theorem gcd_pow_left_dvd_pow_gcd {a b : α} {k : ℕ} : gcd (a ^ k) b ∣ (gcd a b) ^ k :=
+by { rw [gcd_comm, gcd_comm a b], exact gcd_pow_right_dvd_pow_gcd }
+
+private theorem dvd_pow_of_mul_eq_pow {a b c d₁ d₂ : α} (ha : a ≠ 0)
```

Comment:
```suggestion
theorem pow_dvd_of_mul_eq_pow {a b c d₁ d₂ : α} (ha : a ≠ 0)
```
If there is a 0.05% chance that someone might want to reuse the theorem, we tend to make it public (so not `private`).
But giving this a good name seems nontrivial...